src/System.ServiceModel.Syndication/tests/BasicScenarioTests.cs

```diff
@@ -368,43 +358,130 @@ public static void AtomEntryPositiveTest_write()
+        }
+
+        [Fact]
+        public static void RssFeedPositiveTest()
+        {
+            string dataFile = @"rss_feeds.dat";
             List<string> fileList = GetTestFilesForFeedTest(dataFile);
+            List<AllowableDifference> allowableDifferences = GetRssFeedPositiveTestAllowableDifferences();
 
             foreach (string file in fileList)
             {
-                AtomFeedPositiveTest(file);
+                FeedPositiveTest(file, FeedFormatterType.Rss20FeedFormatter, allowableDifferences);
             }
         }
 
-        private static void AtomFeedPositiveTest(string file)
+        private static void EntryPositiveTest(string file, SyndicationItemFormatterType syndicationItemFormatterType)
+        {
+
+            string serializeFilePath = Path.GetTempFileName();
+            try
+            {
+                SyndicationItem feedObjct = null;
+                using (FileStream fileStream = new FileStream(file, FileMode.Open, FileAccess.Read))
+                {
+                    XmlReader reader = XmlDictionaryReader.CreateTextReader(fileStream, XmlDictionaryReaderQuotas.Max);
+                    feedObjct = SyndicationItem.Load(reader);
+                    reader.Close();
+                }
+
+                using (FileStream fileStream = new FileStream(serializeFilePath, FileMode.OpenOrCreate))
+                {
+                    XmlWriter writer = XmlDictionaryWriter.CreateTextWriter(fileStream);
+                    SyndicationItemFormatter formatter = null;
+
+                    if (syndicationItemFormatterType == SyndicationItemFormatterType.Atom10ItemFormatter)
+                    {
+                        formatter = new Atom10ItemFormatter(feedObjct);
+                    }
+                    else if (syndicationItemFormatterType == SyndicationItemFormatterType.Rss20ItemFormatter)
+                    {
+                        formatter = new Rss20ItemFormatter(feedObjct);
+                    }
+                    else
+                    {
+                        throw new ArgumentOutOfRangeException("syndicationItemFormatterType");
+                    }
+
+                    formatter.WriteTo(writer);
+                    writer.Flush();
+                    writer.Close();
+                }
+
+                // compare file filePath and serializeFilePath
+                XmlDiff diff = new XmlDiff()
+                {
+                    Option = XmlDiffOption.IgnoreComments | XmlDiffOption.IgnorePrefix | XmlDiffOption.IgnoreWhitespace | XmlDiffOption.IgnoreChildOrder | XmlDiffOption.IgnoreAttributeOrder
+                };
+
+                Assert.True(diff.Compare(file, serializeFilePath));
+            }
+            catch(Exception e)
+            {
+                Exception newEx = new Exception($"Failed File Name:{file}", e);
+                throw newEx;
+            }
+            finally
+            {
+                File.Delete(serializeFilePath);
```

Comment:
I found it's easier for debugging if we keep the temp file when the test failed. Can you please have a look at the method below and delete the file only when the test passes?

Reply:
Done.